pkg/cmd/pulumi/util_remote.go

```diff
@@ -406,7 +411,7 @@ func runDeployment(ctx context.Context, cmd *cobra.Command, opts display.Options
 
 	// Ensure the cloud backend is being used.
 	cb, isCloud := b.(httpstate.Backend)
-	if !isCloud {
+	if !isCloud && backendInstance == nil /* mock backend */ {
```

Comment:
```suggestion
	isMockBackend := backendInstance != nil
	if !isCloud && !isMockBackend {
```

I struggled to make sense to what the "mock backend" is referring to here for a bit.  I think this would make it easier to read.